### PR TITLE
Create disallow-proc-self-fd-workingdir.yaml

### PR DESCRIPTION
**Pull request by lobuhi**
Mitigation for CVE-2024-21626

## Checklist

- [] I have read the [policy contribution guidelines](https://github.com/kyverno/policies/blob/main/README.md#contribution).
- [] I have added test manifests and resources covering both positive and negative tests that prove this policy works as intended.
- [] I have added the artifacthub-pkg.yml file and have verified it is complete and correct.

### CODE REVIEW

File: pod-security/baseline/disallow-proc-self-fd-workingdir/disallow-proc-self-fd-workingdir.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: disallow-proc-self-fd-workingdir
```
Comment (JimBugwadia) [inline pod-security/baseline/disallow-proc-self-fd-workingdir/disallow-proc-self-fd-workingdir.yaml:4]:
```suggestion
  name: disallow-proc-self-fd-workingdir
  annotations:
    policies.kyverno.io/title: Disallow workingDir with self file descriptors
    policies.kyverno.io/category: Security
    policies.kyverno.io/severity: Critical
    policies.kyverno.io/subject: Pod
    kyverno.io/kyverno-version: 1.11.4
    kyverno.io/kubernetes-version: "1.28"
    policies.kyverno.io/description: >-
      This policy prevents the configuration of a container workingDir with file descriptors
      that start with /proc/self/fd. This is a partial mitigation for CVE-2024-21626 aka
      Leaky Vessels Container Breakout. For additional details and mitigation please refer to
      https://snyk.io/blog/leaky-vessels-docker-runc-container-breakout-vulnerabilities/.
```

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
@lobuhi - thanks for the contribution! I added the annotations required for the policy to be rendered on the website. Please review and update or accept.

**Review by JimBugwadia — CHANGES_REQUESTED** [review]
Apologies, but I did not notice this earlier! 

The `pod-security/baseline/` directory is for PSS policies baseline profile:

https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline

Can you please create a new directory for this policy, under `/other`?

See /policies/other/prevent-cr8escape/prevent-cr8escape/ as an example.

**Review by JimBugwadia — COMMENTED** [review]
We also need an `artifacthub-pkg.yml` file in the directory for it to be published here:

   https://artifacthub.io/

Here is an example:  

https://github.com/kyverno/policies/blob/main/other/prevent-cr8escape/artifacthub-pkg.yml

We can add this in a separate PR if you prefer.

**Review by chipzoller — CHANGES_REQUESTED** [review]
Checklist has not been followed per the PR template.

**Comment by lobuhi** [comment]
I will close this, sorry! 
New PR https://github.com/kyverno/policies/pull/898

**Pull request closed without merging**